MaterialLib/MPL/Properties/PermeabilityMohrCoulombFailureIndexModel.cpp

```diff
@@ -0,0 +1,155 @@
+        std::string name, ParameterLib::Parameter<double> const& k0,
+        double const kr, double const b, double const c, double const phi,
+        double const k_max,
+        ParameterLib::CoordinateSystem const* const local_coordinate_system)
+    : k0_(k0),
+      kr_(kr),
+      b_(b),
+      c_(c),
+      phi_(boost::math::constants::degree<double>() * phi),
+      k_max_(k_max),
+      local_coordinate_system_(local_coordinate_system)
+{
+    name_ = std::move(name);
+}
+
+template <int DisplacementDim>
+void PermeabilityMohrCoulombFailureIndexModel<DisplacementDim>::checkScale()
+    const
+{
+    if (!std::holds_alternative<Medium*>(scale_))
+    {
+        OGS_FATAL(
+            "The property 'PermeabilityMohrCoulombFailureIndexModel' is "
+            "implemented on the 'medium' scale only.");
+    }
+}
+
+template <int DisplacementDim>
+PropertyDataType
+PermeabilityMohrCoulombFailureIndexModel<DisplacementDim>::value(
+    VariableArray const& variable_array,
+    ParameterLib::SpatialPosition const& pos, double const t,
+    double const /*dt*/) const
+{
+    constexpr int symmetric_tensor_size =
+        MathLib::KelvinVector::KelvinVectorDimensions<DisplacementDim>::value;
+    using SymmetricTensor = Eigen::Matrix<double, symmetric_tensor_size, 1>;
+
+    auto const& stress_vector = std::get<SymmetricTensor>(
+        variable_array[static_cast<int>(Variable::stress)]);
+    auto const& stress_tensor = symmetricTensorToFullTensor(stress_vector);
+
+    Eigen::SelfAdjointEigenSolver<Eigen::Matrix<double, 3, 3>>
+        eigenvalue_solver(stress_tensor);
+
+    // Principle stress
+    auto const sigma = eigenvalue_solver.eigenvalues();
+
+    auto k_data = k0_(t, pos);
+
+    double const max_sigma = std::max(std::fabs(sigma[0]), std::fabs(sigma[2]));
+
+    if (max_sigma < std::numeric_limits<double>::epsilon())
+    {
+        return fromVector(k_data);
+    }
+
+    double const sigma_m = 0.5 * (sigma[2] + sigma[0]);
+    double tau_tt = c_ * std::cos(phi_) - sigma_m * std::sin(phi_);
+    const double apex_cut_offset = 0.001;
+    if (std::fabs(tau_tt) / max_sigma < apex_cut_offset)
+    {
+        tau_tt = apex_cut_offset * c_ * std::cos(phi_);
+    }
+
+    double const f = 0.5 * (sigma[2] - sigma[0]) / tau_tt;
+
+    if (f >= 1.0)
+    {
+        for (auto& k_i : k_data)
+        {
+            const double exp_value = std::exp(b_ * f);
+            if (std::isinf(exp_value))
+            {
+                k_i = k_max_;
+            }
+
+            k_i += kr_ * exp_value;
+            if (k_i > k_max_)
+            {
+                k_i = k_max_;
+            }
+        }
+    }
+
+    // Local coordinate transformation is only applied for the case that the
+    // initial intrinsic permeability is given with orthotropic assumption.
+    if (local_coordinate_system_ &&
+        ((k_data.size() == 2) || (k_data.size() == 3)))
+    {
+        assert(k_data.size() == DisplacementDim);
```

Comment:
assert condition is always fulfilled, so moving it to the if condition would yield:
```suggestion
    if (local_coordinate_system_ && k_data.size() == DisplacementDim)
    {
```
which is simpler.

Reply:
Took it.